# BAX 443 - Analytical Decision Making - HW1

Pallavi Sama

Analytical Decision Making, MSBA, UC Davis

May 1st, 2018

<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Exercise-1-:-Online-Advertising-Sensitivity-Analyses" data-toc-modified-id="Exercise-1-:-Online-Advertising-Sensitivity-Analyses-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Exercise 1 : Online Advertising Sensitivity Analyses</a></span><ul class="toc-item"><li><span><a href="#Decision-variables" data-toc-modified-id="Decision-variables-1.1"><span class="toc-item-num">1.1&nbsp;&nbsp;</span>Decision variables</a></span></li><li><span><a href="#Objective-function" data-toc-modified-id="Objective-function-1.2"><span class="toc-item-num">1.2&nbsp;&nbsp;</span>Objective function</a></span></li><li><span><a href="#Constraints" data-toc-modified-id="Constraints-1.3"><span class="toc-item-num">1.3&nbsp;&nbsp;</span>Constraints</a></span></li><li><span><a href="#Optimal-Solution" data-toc-modified-id="Optimal-Solution-1.4"><span class="toc-item-num">1.4&nbsp;&nbsp;</span>Optimal Solution</a></span></li><li><span><a href="#Sensitivity-Analysis" data-toc-modified-id="Sensitivity-Analysis-1.5"><span class="toc-item-num">1.5&nbsp;&nbsp;</span>Sensitivity Analysis</a></span><ul class="toc-item"><li><span><a href="#Geico-Sports" data-toc-modified-id="Geico-Sports-1.5.1"><span class="toc-item-num">1.5.1&nbsp;&nbsp;</span>Geico Sports</a></span></li><li><span><a href="#Geico-National" data-toc-modified-id="Geico-National-1.5.2"><span class="toc-item-num">1.5.2&nbsp;&nbsp;</span>Geico National</a></span></li><li><span><a href="#Delta-Sports" data-toc-modified-id="Delta-Sports-1.5.3"><span class="toc-item-num">1.5.3&nbsp;&nbsp;</span>Delta Sports</a></span></li><li><span><a href="#Delta-National" data-toc-modified-id="Delta-National-1.5.4"><span class="toc-item-num">1.5.4&nbsp;&nbsp;</span>Delta National</a></span></li><li><span><a href="#T-Mobile-Sports" data-toc-modified-id="T-Mobile-Sports-1.5.5"><span class="toc-item-num">1.5.5&nbsp;&nbsp;</span>T-Mobile Sports</a></span></li><li><span><a href="#T-Mobile-National" data-toc-modified-id="T-Mobile-National-1.5.6"><span class="toc-item-num">1.5.6&nbsp;&nbsp;</span>T-Mobile National</a></span></li><li><span><a href="#CapitalOne-Sports" data-toc-modified-id="CapitalOne-Sports-1.5.7"><span class="toc-item-num">1.5.7&nbsp;&nbsp;</span>CapitalOne Sports</a></span></li><li><span><a href="#CapitalOne-National" data-toc-modified-id="CapitalOne-National-1.5.8"><span class="toc-item-num">1.5.8&nbsp;&nbsp;</span>CapitalOne National</a></span></li></ul></li><li><span><a href="#Results/-Observations/-Recommendations" data-toc-modified-id="Results/-Observations/-Recommendations-1.6"><span class="toc-item-num">1.6&nbsp;&nbsp;</span>Results/ Observations/ Recommendations</a></span></li></ul></li><li><span><a href="#Exercise-2-:-Operations-Excellence" data-toc-modified-id="Exercise-2-:-Operations-Excellence-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Exercise 2 : Operations Excellence</a></span><ul class="toc-item"><li><span><a href="#For-separate-optimization-on-Plant-A-and-B" data-toc-modified-id="For-separate-optimization-on-Plant-A-and-B-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>For separate optimization on Plant A and B</a></span><ul class="toc-item"><li><span><a href="#Decision-variables" data-toc-modified-id="Decision-variables-2.1.1"><span class="toc-item-num">2.1.1&nbsp;&nbsp;</span>Decision variables</a></span></li><li><span><a href="#Objective-function" data-toc-modified-id="Objective-function-2.1.2"><span class="toc-item-num">2.1.2&nbsp;&nbsp;</span>Objective function</a></span></li><li><span><a href="#Constraints" data-toc-modified-id="Constraints-2.1.3"><span class="toc-item-num">2.1.3&nbsp;&nbsp;</span>Constraints</a></span></li><li><span><a href="#Solution-for-Plant-A" data-toc-modified-id="Solution-for-Plant-A-2.1.4"><span class="toc-item-num">2.1.4&nbsp;&nbsp;</span>Solution for Plant A</a></span></li><li><span><a href="#Solution-for-Plant-B" data-toc-modified-id="Solution-for-Plant-B-2.1.5"><span class="toc-item-num">2.1.5&nbsp;&nbsp;</span>Solution for Plant B</a></span></li></ul></li><li><span><a href="#Solution-for-Joint-optimization" data-toc-modified-id="Solution-for-Joint-optimization-2.2"><span class="toc-item-num">2.2&nbsp;&nbsp;</span>Solution for Joint optimization</a></span><ul class="toc-item"><li><span><a href="#Decision-variables" data-toc-modified-id="Decision-variables-2.2.1"><span class="toc-item-num">2.2.1&nbsp;&nbsp;</span>Decision variables</a></span></li><li><span><a href="#Objective-function" data-toc-modified-id="Objective-function-2.2.2"><span class="toc-item-num">2.2.2&nbsp;&nbsp;</span>Objective function</a></span></li><li><span><a href="#Constraints" data-toc-modified-id="Constraints-2.2.3"><span class="toc-item-num">2.2.3&nbsp;&nbsp;</span>Constraints</a></span></li></ul></li><li><span><a href="#Results/-Observations/-Recommendations" data-toc-modified-id="Results/-Observations/-Recommendations-2.3"><span class="toc-item-num">2.3&nbsp;&nbsp;</span>Results/ Observations/ Recommendations</a></span></li></ul></li><li><span><a href="#Exercise-3-:-Hotel-La-Quinta-Motor-Inns-(LQM-Profitability-Analysis)" data-toc-modified-id="Exercise-3-:-Hotel-La-Quinta-Motor-Inns-(LQM-Profitability-Analysis)-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Exercise 3 : Hotel La Quinta Motor Inns (LQM Profitability Analysis)</a></span><ul class="toc-item"><li><span><a href="#Question-1" data-toc-modified-id="Question-1-3.1"><span class="toc-item-num">3.1&nbsp;&nbsp;</span>Question 1</a></span></li><li><span><a href="#Question-2" data-toc-modified-id="Question-2-3.2"><span class="toc-item-num">3.2&nbsp;&nbsp;</span>Question 2</a></span></li><li><span><a href="#Question-3" data-toc-modified-id="Question-3-3.3"><span class="toc-item-num">3.3&nbsp;&nbsp;</span>Question 3</a></span></li><li><span><a href="#Question-4" data-toc-modified-id="Question-4-3.4"><span class="toc-item-num">3.4&nbsp;&nbsp;</span>Question 4</a></span></li><li><span><a href="#Question-5" data-toc-modified-id="Question-5-3.5"><span class="toc-item-num">3.5&nbsp;&nbsp;</span>Question 5</a></span></li><li><span><a href="#Results/-Observations/-Recommendations" data-toc-modified-id="Results/-Observations/-Recommendations-3.6"><span class="toc-item-num">3.6&nbsp;&nbsp;</span>Results/ Observations/ Recommendations</a></span></li></ul></li></ul></div>

### Exercise 1 : Online Advertising Sensitivity Analyses

#### Decision variables
Let $i={1,2,3,4}$ be the index for the advertisers such that
* $i=1$ is GEICO
* $i=2$ is Delta
* $i=3$ is T-Mobile
* $i=4$ is Capital One

Let $j={1,2}$ be the index for the category, such as $j=1$ is the Sports categorty and $j=2$ is the National category.

So $x_{11}$ is the number of impressions for Geico in Sports and $x_{12}$ is the number of impressions for GEICO in the National category

#### Objective function
The objective is to maximize the advertising revenues. These revenues are driven by the cost per click charged to the advertisers, i.e., \$2.4 and the click-through rates.

Let $\kappa_{ij}$ be the CTR of advertiser $i$ in category $j$. For instance, the CTR of T-Mobile in the sports category is $\kappa_{31}=1.0\%$, whereas its CTR in the National category is $\kappa_{32}=3.0\%$.

The objective function is thus:
\begin{equation}
Objective=2.4\times \sum_{i=1}^4\sum_{j=1}^{2}  \kappa_{ij}\times x_{ij}
\end{equation}

#### Constraints
The first table provides the following constraints:
\begin{align}
x_{11}&\geq 2,000,000\\
x_{12}&\geq 1,000,000\\
x_{22}&\geq 1,000,000\\
x_{21}+x_{22}&\geq 2,000,000\\
x_{31}&\geq 1,000,000\\
x_{32}&\geq 1,000,000\\
x_{31}+x_{32}&\geq 3,000,000\\
x_{41}+x_{42}&\geq 2,000,000\\
\end{align}

"Assume that the Sports section gets six million views per day and the National section
get five million views per day"
This statement gives two constraints:
\begin{align}
\sum_{i=1}^{4}x_{i1}&\leq 6,000,000\\
\sum_{i=1}^{4}x_{i2}&\leq 5,000,000\\
\end{align}

Non-negativity constraints are:
\begin{equation}
x_{ij}\geq 0
\end{equation}

#### Optimal Solution

In [46]:
#Uncomment the below 2 pip commands when running on google colab
#!pip install cvxpy
#!pip install cvxopt
#Load required packages
import numpy 
import math
import cvxpy as cvx
from numpy import *
from cvxpy import *
import pandas as pd

#Define variables
kappa=matrix([[2.5,2.0,1.0,1.5],[0.8,1.0,3.0,2.0]])
kappa1=kappa[0]
kappa2=kappa[1]
x1=cvx.Int(4,1)
x2=cvx.Int(4,1)

# Objective Function
Z1=sum_entries(kappa1*x1)*2.4/100
Z2=sum_entries(kappa2*x2)*2.4/100
objective = cvx.Maximize((Z1+Z2))

# Constraints
c1=(sum_entries(x1))<=6*1000000 # Capacity on Sports
c2=(sum_entries(x2))<=5*1000000 # Capacity on Sports
c3=x1[0]>=2*1000000 # Geico must have at least 2m in sports
c4=x2[0]>=1*1000000 # Geico must have at least 1m in sports
c5=x2[1]>=1*1000000 # Delta must have at least 1m in sports
c6=x1[1]+x2[1]==2*1000000 # Delta's total # of impressions must be equal to 2m total
c7=x1[2]>=1*1000000# Tmobile must have at least 1m impression in sports
c8=x2[2]>=1*1000000# Tmobile must have at least 1m impression in national
c9=x1[2]+x2[2]==3*1000000 # Tmobile's total # of impressions must be equal 3m total
c10=x1[3]+x2[3]==2*1000000# Capital One's impression equal 2m
c=[c1,c2,c3,c4,c5,c6,c7,c8,c9,c10,x1>=0,x2>=0]

#Solution
prob = cvx.Problem(objective, c)
result = prob.solve()
print('The optimal revenue is ')
print(prob.value)
print('The optimal allocation in the sports category is')
print(x1.value)
print('The optimal allocation in the national category is')
print(x2.value)

The optimal revenue is 
523199.99999759113
The optimal allocation in the sports category is
[[2999999.99999905]
 [ 999999.99995182]
 [1000000.00001714]
 [1000000.0000317 ]]
The optimal allocation in the national category is
[[1000000.0000004 ]
 [1000000.00004818]
 [1999999.99998286]
 [ 999999.9999683 ]]


#### Sensitivity Analysis

##### Geico Sports

In [2]:
#Define variables
x1=cvx.Int(4,1)
x2=cvx.Int(4,1)

#Constraints
c1=(sum_entries(x1))<=6*1000000#Capacity on Sports
c2=(sum_entries(x2))<=5*1000000#Capacity on Sports
c3=x1[0]>=2*1000000 # Geico must have at least 2m in sports
c4=x2[0]>=1*1000000 # Geico must have at least 1m in sports
c5=x2[1]>=1*1000000 # Delta must have at least 1m in sports
c6=x1[1]+x2[1]==2*1000000 # Delta's total # of impressions must be equal to 2m total
c7=x1[2]>=1*1000000# Tmobile must have at least 1m impression in sports
c8=x2[2]>=1*1000000# Tmobile must have at least 1m impression in national
c9=x1[2]+x2[2]==3*1000000 # Tmobile's total # of impressions must be equal 3m total
c10=x1[3]+x2[3]==2*1000000# Capital One's impression equal 2m
c=[c1,c2,c3,c4,c5,c6,c7,c8,c9,c10,x1>=0,x2>=0]

#Change click-thru rates here to run sensitvity analysis
arr_i = numpy.arange(2.0, 3.1, 0.1)
geico_s = pd.DataFrame()
for i in range(0, len(arr_i)):
    kappa=matrix([[arr_i[i],2.0,1.0,1.5],[0.8,1.0,3.0,2.0]])
    kappa1=kappa[0]
    kappa2=kappa[1]
        
    #Objective Function
    Z1=sum_entries(kappa1*x1)*2.4/100
    Z2=sum_entries(kappa2*x2)*2.4/100
    objective = cvx.Maximize((Z1+Z2))
        
    #Solution
    prob = cvx.Problem(objective, c)
    result = prob.solve()
    print('######################## GEICO - Sports CTR as %s%% #######################'%arr_i[i])
    print('The optimal revenue is')
    print(prob.value)
    print('The optimal allocation in the sports category is')
    print(x1.value)
    print('The optimal allocation in the national category is')
    print(x2.value)
        
    #Appending to matrix
    geico_s = geico_s.append([[arr_i[i],prob.value]]) 

######################## GEICO - Sports CTR as 2.0% #######################
The optimal revenue is
487199.9999981021
The optimal allocation in the sports category is
[[2999999.99999848]
 [ 999999.99998563]
 [1000000.00003637]
 [ 999999.99997915]]
The optimal allocation in the national category is
[[1000000.00000082]
 [1000000.00001437]
 [1999999.99996362]
 [1000000.00002085]]
######################## GEICO - Sports CTR as 2.1% #######################
The optimal revenue is
494399.99999717786
The optimal allocation in the sports category is
[[2999999.99999857]
 [ 999999.99998237]
 [1000000.00005878]
 [ 999999.9999599 ]]
The optimal allocation in the national category is
[[1000000.0000007 ]
 [1000000.00001763]
 [1999999.99994122]
 [1000000.0000401 ]]
######################## GEICO - Sports CTR as 2.2% #######################
The optimal revenue is
501599.9999970976
The optimal allocation in the sports category is
[[2999999.9999989 ]
 [ 999999.99998037]
 [1000000.00005939]
 [ 999999.99996

In [3]:
#To view the results in a table - sorted the table by top to bottom Revenue 
geico_s.columns=['Sports CTR','Optimal Revenue']
geico_s.sort_values(by=['Optimal Revenue'],ascending=False)

Sports CTR  Optimal Revenue
0         3.0    559199.999998
0         2.9    551999.999998
0         2.8    544799.999997
0         2.7    537599.999997
0         2.6    530399.999997
0         2.5    523199.999998
0         2.4    515999.999998
0         2.3    508799.999998
0         2.2    501599.999997
0         2.1    494399.999997
0         2.0    487199.999998

In [4]:
#To check the variance in the revenue
print(geico_s['Optimal Revenue'].mean())
print(geico_s['Optimal Revenue'].std())

523199.9999974903
23879.698490609273


##### Geico National

In [5]:
#Define variables
x1=cvx.Int(4,1)
x2=cvx.Int(4,1)

#Constraints
c1=(sum_entries(x1))<=6*1000000#Capacity on Sports
c2=(sum_entries(x2))<=5*1000000#Capacity on Sports
c3=x1[0]>=2*1000000 # Geico must have at least 2m in sports
c4=x2[0]>=1*1000000 # Geico must have at least 1m in sports
c5=x2[1]>=1*1000000 # Delta must have at least 1m in sports
c6=x1[1]+x2[1]==2*1000000 # Delta's total # of impressions must be equal to 2m total
c7=x1[2]>=1*1000000# Tmobile must have at least 1m impression in sports
c8=x2[2]>=1*1000000# Tmobile must have at least 1m impression in national
c9=x1[2]+x2[2]==3*1000000 # Tmobile's total # of impressions must be equal 3m total
c10=x1[3]+x2[3]==2*1000000# Capital Obe's impression equal 2m
c=[c1,c2,c3,c4,c5,c6,c7,c8,c9,c10,x1>=0,x2>=0]

#Change click-thru rates here to run sensitvity analysis
arr_i = numpy.arange(0.1, 1.2, 0.1)
geico_n = pd.DataFrame()
for i in range(0, len(arr_i)):
    kappa=matrix([[2.5,2.0,1.0,1.5],[arr_i[i],1.0,3.0,2.0]])
    kappa1=kappa[0]
    kappa2=kappa[1]
        
    #Objective Function
    Z1=sum_entries(kappa1*x1)*2.4/100
    Z2=sum_entries(kappa2*x2)*2.4/100
    objective = cvx.Maximize((Z1+Z2))
        
    #Solution
    prob = cvx.Problem(objective, c)
    result = prob.solve()
    print('######################## GEICO - National CTR as %s%% #######################'%arr_i[i])
    print('The optimal revenue is')
    print(prob.value)
    print('The optimal allocation in the sports category is')
    print(x1.value)
    print('The optimal allocation in the national category is')
    print(x2.value)
        
    #Appending to matrix
    geico_n = geico_n.append([[arr_i[i],prob.value]]) 

######################## GEICO - National CTR as 0.1% #######################
The optimal revenue is
506399.99999937543
The optimal allocation in the sports category is
[[2999999.99998837]
 [ 999999.9999969 ]
 [ 999999.99999584]
 [1000000.0000027 ]]
The optimal allocation in the national category is
[[ 999999.99999176]
 [1000000.0000031 ]
 [2000000.00000416]
 [ 999999.9999973 ]]
######################## GEICO - National CTR as 0.2% #######################
The optimal revenue is
508799.99999935593
The optimal allocation in the sports category is
[[2999999.99998838]
 [ 999999.9999969 ]
 [ 999999.99999583]
 [1000000.0000027 ]]
The optimal allocation in the national category is
[[ 999999.99999172]
 [1000000.0000031 ]
 [2000000.00000417]
 [ 999999.9999973 ]]
######################## GEICO - National CTR as 0.30000000000000004% #######################
The optimal revenue is
511199.9999993365
The optimal allocation in the sports category is
[[2999999.99998839]
 [ 999999.99999689]
 [ 999999.99

In [6]:
#To view the results in a table - sorted the table by top to bottom Revenue 
geico_n.columns=['National CTR','Optimal Revenue']
geico_n.sort_values(by=['Optimal Revenue'],ascending=False)

National CTR  Optimal Revenue
0           1.1    530399.999998
0           1.0    527999.999998
0           0.9    525599.999998
0           0.8    523199.999998
0           0.7    520799.999997
0           0.6    518399.999996
0           0.5    515999.999999
0           0.4    513599.999999
0           0.3    511199.999999
0           0.2    508799.999999
0           0.1    506399.999999

In [7]:
#To check the variance in the revenue
print(geico_n['Optimal Revenue'].mean())
print(geico_n['Optimal Revenue'].std())

518399.9999981961
7959.899496102553


##### Delta Sports

In [8]:
#Define variables
x1=cvx.Int(4,1)
x2=cvx.Int(4,1)

#Constraints
c1=(sum_entries(x1))<=6*1000000#Capacity on Sports
c2=(sum_entries(x2))<=5*1000000#Capacity on Sports
c3=x1[0]>=2*1000000 # Geico must have at least 2m in sports
c4=x2[0]>=1*1000000 # Geico must have at least 1m in sports
c5=x2[1]>=1*1000000 # Delta must have at least 1m in sports
c6=x1[1]+x2[1]==2*1000000 # Delta's total # of impressions must be equal to 2m total
c7=x1[2]>=1*1000000# Tmobile must have at least 1m impression in sports
c8=x2[2]>=1*1000000# Tmobile must have at least 1m impression in national
c9=x1[2]+x2[2]==3*1000000 # Tmobile's total # of impressions must be equal 3m total
c10=x1[3]+x2[3]==2*1000000# Capital Obe's impression equal 2m
c=[c1,c2,c3,c4,c5,c6,c7,c8,c9,c10,x1>=0,x2>=0]

#Change click-thru rates here to run sensitvity analysis
arr_i = numpy.arange(1.5, 2.6, 0.1)
delta_s = pd.DataFrame()
for i in range(0, len(arr_i)):
    kappa=matrix([[2.5,arr_i[i],1.0,1.5],[0.8,1.0,3.0,2.0]])
    kappa1=kappa[0]
    kappa2=kappa[1]
        
    #Objective Function
    Z1=sum_entries(kappa1*x1)*2.4/100
    Z2=sum_entries(kappa2*x2)*2.4/100
    objective = cvx.Maximize((Z1+Z2))
        
    #Solution
    prob = cvx.Problem(objective, c)
    result = prob.solve()
    print('######################## DELTA - Sports CTR as %s%% #######################'%arr_i[i])
    print('The optimal revenue is')
    print(prob.value)
    print('The optimal allocation in the sports category is')
    print(x1.value)
    print('The optimal allocation in the national category is')
    print(x2.value)
        
    #Appending to matrix
    delta_s = delta_s.append([[arr_i[i],prob.value]]) 

######################## DELTA - Sports CTR as 1.5% #######################
The optimal revenue is
511199.9999979987
The optimal allocation in the sports category is
[[2999999.9999822 ]
 [ 999999.99997911]
 [1000000.00001169]
 [1000000.000021  ]]
The optimal allocation in the national category is
[[1000000.00000681]
 [1000000.00002089]
 [1999999.99998831]
 [ 999999.999979  ]]
######################## DELTA - Sports CTR as 1.6% #######################
The optimal revenue is
513599.99999669514
The optimal allocation in the sports category is
[[2999999.99998987]
 [1000000.        ]
 [1000000.00007541]
 [ 999999.99993273]]
The optimal allocation in the national category is
[[1000000.00000601]
 [1000000.        ]
 [1999999.99992459]
 [1000000.00006727]]
######################## DELTA - Sports CTR as 1.7000000000000002% #######################
The optimal revenue is
515999.9999980414
The optimal allocation in the sports category is
[[2999999.99998495]
 [ 999999.99997934]
 [1000000.00001253]


In [9]:
#To view the results in a table - sorted the table by top to bottom Revenue 
delta_s.columns=['Sports CTR','Optimal Revenue']
delta_s.sort_values(by=['Optimal Revenue'],ascending=False)

Sports CTR  Optimal Revenue
0         2.5    535199.999999
0         2.4    532799.999999
0         2.3    530399.999998
0         2.2    527999.999997
0         2.1    525599.999997
0         2.0    523199.999998
0         1.9    520799.999999
0         1.8    518399.999999
0         1.7    515999.999998
0         1.6    513599.999997
0         1.5    511199.999998

In [10]:
#To check the variance in the revenue
print(delta_s['Optimal Revenue'].mean())
print(delta_s['Optimal Revenue'].std())

523199.9999981152
7959.8994971509765


##### Delta National

In [11]:
#Define variables
x1=cvx.Int(4,1)
x2=cvx.Int(4,1)

#Constraints
c1=(sum_entries(x1))<=6*1000000#Capacity on Sports
c2=(sum_entries(x2))<=5*1000000#Capacity on Sports
c3=x1[0]>=2*1000000 # Geico must have at least 2m in sports
c4=x2[0]>=1*1000000 # Geico must have at least 1m in sports
c5=x2[1]>=1*1000000 # Delta must have at least 1m in sports
c6=x1[1]+x2[1]==2*1000000 # Delta's total # of impressions must be equal to 2m total
c7=x1[2]>=1*1000000# Tmobile must have at least 1m impression in sports
c8=x2[2]>=1*1000000# Tmobile must have at least 1m impression in national
c9=x1[2]+x2[2]==3*1000000 # Tmobile's total # of impressions must be equal 3m total
c10=x1[3]+x2[3]==2*1000000# Capital Obe's impression equal 2m
c=[c1,c2,c3,c4,c5,c6,c7,c8,c9,c10,x1>=0,x2>=0]

#Change click-thru rates here to run sensitvity analysis
arr_i = numpy.arange(0.5, 1.6, 0.1)
delta_n = pd.DataFrame()
for i in range(0, len(arr_i)):
    kappa=matrix([[2.5,2.0,1.0,1.5],[0.8,arr_i[i],3.0,2.0]])
    kappa1=kappa[0]
    kappa2=kappa[1]
        
    #Objective Function
    Z1=sum_entries(kappa1*x1)*2.4/100
    Z2=sum_entries(kappa2*x2)*2.4/100
    objective = cvx.Maximize((Z1+Z2))
        
    #Solution
    prob = cvx.Problem(objective, c)
    result = prob.solve()
    print('######################## DELTA - National CTR as %s%% #######################'%arr_i[i])
    print('The optimal revenue is')
    print(prob.value)
    print('The optimal allocation in the sports category is')
    print(x1.value)
    print('The optimal allocation in the national category is')
    print(x2.value)
        
    #Appending to matrix
    delta_n = delta_n.append([[arr_i[i],prob.value]]) 

######################## DELTA - National CTR as 0.5% #######################
The optimal revenue is
511199.99999933795
The optimal allocation in the sports category is
[[2999999.99999898]
 [ 999999.99999752]
 [1000000.00001337]
 [ 999999.99998987]]
The optimal allocation in the national category is
[[1000000.00000045]
 [1000000.00000248]
 [1999999.99998663]
 [1000000.00001013]]
######################## DELTA - National CTR as 0.6% #######################
The optimal revenue is
513599.99999911885
The optimal allocation in the sports category is
[[2999999.99999906]
 [ 999999.99999225]
 [1000000.00001309]
 [ 999999.99999538]]
The optimal allocation in the national category is
[[1000000.00000044]
 [1000000.00000775]
 [1999999.99998691]
 [1000000.00000462]]
######################## DELTA - National CTR as 0.7% #######################
The optimal revenue is
515999.9999982609
The optimal allocation in the sports category is
[[2999999.99999885]
 [ 999999.99996768]
 [1000000.0000076 ]
 [100000

In [12]:
#To view the results in a table - sorted the table by top to bottom Revenue 
delta_n.columns=['National CTR','Optimal Revenue']
delta_n.sort_values(by=['Optimal Revenue'],ascending=False)

National CTR  Optimal Revenue
0           1.5    535199.999998
0           1.4    532799.999997
0           1.3    530399.999998
0           1.2    527999.999999
0           1.1    525599.999999
0           1.0    523199.999998
0           0.9    520799.999997
0           0.8    518399.999997
0           0.7    515999.999998
0           0.6    513599.999999
0           0.5    511199.999999

In [13]:
#To check the variance in the revenue
print(delta_n['Optimal Revenue'].mean())
print(delta_n['Optimal Revenue'].std())

523199.9999981219
7959.899496586239


##### T-Mobile Sports

In [14]:
#Define variables
x1=cvx.Int(4,1)
x2=cvx.Int(4,1)

#Constraints
c1=(sum_entries(x1))<=6*1000000#Capacity on Sports
c2=(sum_entries(x2))<=5*1000000#Capacity on Sports
c3=x1[0]>=2*1000000 # Geico must have at least 2m in sports
c4=x2[0]>=1*1000000 # Geico must have at least 1m in sports
c5=x2[1]>=1*1000000 # Delta must have at least 1m in sports
c6=x1[1]+x2[1]==2*1000000 # Delta's total # of impressions must be equal to 2m total
c7=x1[2]>=1*1000000# Tmobile must have at least 1m impression in sports
c8=x2[2]>=1*1000000# Tmobile must have at least 1m impression in national
c9=x1[2]+x2[2]==3*1000000 # Tmobile's total # of impressions must be equal 3m total
c10=x1[3]+x2[3]==2*1000000# Capital Obe's impression equal 2m
c=[c1,c2,c3,c4,c5,c6,c7,c8,c9,c10,x1>=0,x2>=0]

#Change click-thru rates here to run sensitvity analysis
arr_i = numpy.arange(0.5, 1.6, 0.1)
tmob_s = pd.DataFrame()
for i in range(0, len(arr_i)):
    kappa=matrix([[2.5,2.0,arr_i[i],1.5],[0.8,1.0,3.0,2.0]])
    kappa1=kappa[0]
    kappa2=kappa[1]
        
    #Objective Function
    Z1=sum_entries(kappa1*x1)*2.4/100
    Z2=sum_entries(kappa2*x2)*2.4/100
    objective = cvx.Maximize((Z1+Z2))
        
    #Solution
    prob = cvx.Problem(objective, c)
    result = prob.solve()
    print('######################## T-Mobile - Sports CTR as %s%% #######################'%arr_i[i])
    print('The optimal revenue is')
    print(prob.value)
    print('The optimal allocation in the sports category is')
    print(x1.value)
    print('The optimal allocation in the national category is')
    print(x2.value)
        
    #Appending to matrix
    tmob_s = tmob_s.append([[arr_i[i],prob.value]]) 

######################## T-Mobile - Sports CTR as 0.5% #######################
The optimal revenue is
511199.9999989105
The optimal allocation in the sports category is
[[2999999.99999874]
 [ 999999.99997467]
 [1000000.00000213]
 [1000000.0000241 ]]
The optimal allocation in the national category is
[[1000000.00000058]
 [1000000.00002533]
 [1999999.99999787]
 [ 999999.9999759 ]]
######################## T-Mobile - Sports CTR as 0.6% #######################
The optimal revenue is
513599.99999913125
The optimal allocation in the sports category is
[[2999999.9999993 ]
 [ 999999.99998012]
 [1000000.00000244]
 [1000000.00001791]]
The optimal allocation in the national category is
[[1000000.00000031]
 [1000000.00001988]
 [1999999.99999756]
 [ 999999.99998208]]
######################## T-Mobile - Sports CTR as 0.7% #######################
The optimal revenue is
515999.9999981259
The optimal allocation in the sports category is
[[2999999.99998668]
 [ 999999.99998113]
 [1000000.00000897]
 [1000

In [15]:
#To view the results in a table - sorted the table by top to bottom Revenue 
tmob_s.columns=['Sports CTR','Optimal Revenue']
tmob_s.sort_values(by=['Optimal Revenue'],ascending=False)

Sports CTR  Optimal Revenue
0         1.5    535200.000000
0         1.4    532799.999999
0         1.3    530399.999999
0         1.2    527999.999999
0         1.1    525599.999999
0         1.0    523199.999998
0         0.9    520799.999996
0         0.8    518399.999998
0         0.7    515999.999998
0         0.6    513599.999999
0         0.5    511199.999999

In [16]:
#To check the variance in the revenue
print(tmob_s['Optimal Revenue'].mean())
print(tmob_s['Optimal Revenue'].std())

523199.9999986291
7959.899497206303


##### T-Mobile National

In [17]:
#Define variables
x1=cvx.Int(4,1)
x2=cvx.Int(4,1)

#Constraints
c1=(sum_entries(x1))<=6*1000000#Capacity on Sports
c2=(sum_entries(x2))<=5*1000000#Capacity on Sports
c3=x1[0]>=2*1000000 # Geico must have at least 2m in sports
c4=x2[0]>=1*1000000 # Geico must have at least 1m in sports
c5=x2[1]>=1*1000000 # Delta must have at least 1m in sports
c6=x1[1]+x2[1]==2*1000000 # Delta's total # of impressions must be equal to 2m total
c7=x1[2]>=1*1000000# Tmobile must have at least 1m impression in sports
c8=x2[2]>=1*1000000# Tmobile must have at least 1m impression in national
c9=x1[2]+x2[2]==3*1000000 # Tmobile's total # of impressions must be equal 3m total
c10=x1[3]+x2[3]==2*1000000# Capital Obe's impression equal 2m
c=[c1,c2,c3,c4,c5,c6,c7,c8,c9,c10,x1>=0,x2>=0]

#Change click-thru rates here to run sensitvity analysis
arr_i = numpy.arange(2.5, 3.6, 0.1)
tmob_n = pd.DataFrame()
for i in range(0, len(arr_i)):
    kappa=matrix([[2.5,2.0,1.0,1.5],[0.8,1.0,arr_i[i],2.0]])
    kappa1=kappa[0]
    kappa2=kappa[1]
        
    #Objective Function
    Z1=sum_entries(kappa1*x1)*2.4/100
    Z2=sum_entries(kappa2*x2)*2.4/100
    objective = cvx.Maximize((Z1+Z2))
        
    #Solution
    prob = cvx.Problem(objective, c)
    result = prob.solve()
    print('######################## T-Mobile - National CTR as %s%% #######################'%arr_i[i])
    print('The optimal revenue is')
    print(prob.value)
    print('The optimal allocation in the sports category is')
    print(x1.value)
    print('The optimal allocation in the national category is')
    print(x2.value)
        
    #Appending to matrix
    tmob_n = tmob_n.append([[arr_i[i],prob.value]]) 

######################## T-Mobile - National CTR as 2.5% #######################
The optimal revenue is
499199.9999996031
The optimal allocation in the sports category is
[[2999999.99999875]
 [ 999999.99999762]
 [1000000.00000981]
 [ 999999.99999339]]
The optimal allocation in the national category is
[[1000000.00000048]
 [1000000.00000238]
 [1999999.99999019]
 [1000000.00000661]]
######################## T-Mobile - National CTR as 2.6% #######################
The optimal revenue is
503999.99999919994
The optimal allocation in the sports category is
[[2999999.99998843]
 [ 999999.9999969 ]
 [ 999999.99999574]
 [1000000.00000271]]
The optimal allocation in the national category is
[[ 999999.99999151]
 [1000000.0000031 ]
 [2000000.00000426]
 [ 999999.99999729]]
######################## T-Mobile - National CTR as 2.7% #######################
The optimal revenue is
508799.99999920925
The optimal allocation in the sports category is
[[2999999.99998843]
 [ 999999.9999969 ]
 [ 999999.99999576]

In [18]:
#To view the results in a table - sorted the table by top to bottom Revenue 
tmob_n.columns=['National CTR','Optimal Revenue']
tmob_n.sort_values(by=['Optimal Revenue'],ascending=False)

National CTR  Optimal Revenue
0           3.5    547199.999999
0           3.4    542399.999999
0           3.3    537599.999998
0           3.2    532799.999998
0           3.1    527999.999996
0           3.0    523199.999998
0           2.9    518399.999999
0           2.8    513599.999999
0           2.7    508799.999999
0           2.6    503999.999999
0           2.5    499200.000000

In [19]:
#To check the variance in the revenue
print(tmob_n['Optimal Revenue'].mean())
print(tmob_n['Optimal Revenue'].std())

523199.99999862915
15919.79899335253


##### CapitalOne Sports

In [27]:
#Define variables
x1=cvx.Int(4,1)
x2=cvx.Int(4,1)

#Constraints
c1=(sum_entries(x1))<=6*1000000#Capacity on Sports
c2=(sum_entries(x2))<=5*1000000#Capacity on Sports
c3=x1[0]>=2*1000000 # Geico must have at least 2m in sports
c4=x2[0]>=1*1000000 # Geico must have at least 1m in sports
c5=x2[1]>=1*1000000 # Delta must have at least 1m in sports
c6=x1[1]+x2[1]==2*1000000 # Delta's total # of impressions must be equal to 2m total
c7=x1[2]>=1*1000000# Tmobile must have at least 1m impression in sports
c8=x2[2]>=1*1000000# Tmobile must have at least 1m impression in national
c9=x1[2]+x2[2]==3*1000000 # Tmobile's total # of impressions must be equal 3m total
c10=x1[3]+x2[3]==2*1000000# Capital Obe's impression equal 2m
c=[c1,c2,c3,c4,c5,c6,c7,c8,c9,c10,x1>=0,x2>=0]

#Change click-thru rates here to run sensitvity analysis
arr_i = numpy.arange(1.5, 2.1, 0.1)
cap1_s = pd.DataFrame()
for i in range(0, len(arr_i)):
    kappa=matrix([[2.5,2.0,1.0,arr_i[i]],[0.8,1.0,3.0,2.0]])
    kappa1=kappa[0]
    kappa2=kappa[1]
        
    #Objective Function
    Z1=sum_entries(kappa1*x1)*2.4/100
    Z2=sum_entries(kappa2*x2)*2.4/100
    objective = cvx.Maximize((Z1+Z2))
        
    #Solution
    prob = cvx.Problem(objective, c)
    result = prob.solve()
    print('######################## CapitalOne - Sports CTR as %s%% #######################'%arr_i[i])
    print('The optimal revenue is')
    print(prob.value)
    print('The optimal allocation in the sports category is')
    print(x1.value)
    print('The optimal allocation in the national category is')
    print(x2.value)
        
    #Appending to matrix
    cap1_s = cap1_s.append([[arr_i[i],prob.value]]) 

######################## CapitalOne - Sports CTR as 1.5% #######################
The optimal revenue is
523199.99999759113
The optimal allocation in the sports category is
[[2999999.99999905]
 [ 999999.99995182]
 [1000000.00001714]
 [1000000.0000317 ]]
The optimal allocation in the national category is
[[1000000.0000004 ]
 [1000000.00004818]
 [1999999.99998286]
 [ 999999.9999683 ]]
######################## CapitalOne - Sports CTR as 1.6% #######################
The optimal revenue is
525599.9999992427
The optimal allocation in the sports category is
[[2999999.99998843]
 [ 999999.99999689]
 [ 999999.99999583]
 [1000000.00000266]]
The optimal allocation in the national category is
[[ 999999.99999152]
 [1000000.00000311]
 [2000000.00000417]
 [ 999999.99999734]]
######################## CapitalOne - Sports CTR as 1.7000000000000002% #######################
The optimal revenue is
527999.9999992494
The optimal allocation in the sports category is
[[2999999.99998843]
 [ 999999.9999969 ]
 [ 99

In [28]:
#To view the results in a table - sorted the table by top to bottom Revenue 
cap1_s.columns=['Sports CTR','Optimal Revenue']
cap1_s.sort_values(by=['Optimal Revenue'],ascending=False)[1:]

Sports CTR  Optimal Revenue
0         2.0    535200.000000
0         1.9    532800.000000
0         1.8    530399.999999
0         1.7    527999.999999
0         1.6    525599.999999
0         1.5    523199.999998

In [29]:
#To check the variance in the revenue
print(cap1_s[1:]['Optimal Revenue'].mean())
print(cap1_s[1:]['Optimal Revenue'].std())

531599.9999995962
4489.988864460298


##### CapitalOne National

In [23]:
#Define variables
x1=cvx.Int(4,1)
x2=cvx.Int(4,1)

#Constraints
c1=(sum_entries(x1))<=6*1000000#Capacity on Sports
c2=(sum_entries(x2))<=5*1000000#Capacity on Sports
c3=x1[0]>=2*1000000 # Geico must have at least 2m in sports
c4=x2[0]>=1*1000000 # Geico must have at least 1m in sports
c5=x2[1]>=1*1000000 # Delta must have at least 1m in sports
c6=x1[1]+x2[1]==2*1000000 # Delta's total # of impressions must be equal to 2m total
c7=x1[2]>=1*1000000# Tmobile must have at least 1m impression in sports
c8=x2[2]>=1*1000000# Tmobile must have at least 1m impression in national
c9=x1[2]+x2[2]==3*1000000 # Tmobile's total # of impressions must be equal 3m total
c10=x1[3]+x2[3]==2*1000000# Capital Obe's impression equal 2m
c=[c1,c2,c3,c4,c5,c6,c7,c8,c9,c10,x1>=0,x2>=0]

#Change click-thru rates here to run sensitvity analysis
arr_i = numpy.arange(1.5, 2.6, 0.1)
cap1_n = pd.DataFrame()
for i in range(0, len(arr_i)):
    kappa=matrix([[2.5,2.0,1.0,1.5],[0.8,1.0,3.0,arr_i[i]]])
    kappa1=kappa[0]
    kappa2=kappa[1]
        
    #Objective Function
    Z1=sum_entries(kappa1*x1)*2.4/100
    Z2=sum_entries(kappa2*x2)*2.4/100
    objective = cvx.Maximize((Z1+Z2))
        
    #Solution
    prob = cvx.Problem(objective, c)
    result = prob.solve()
    print('######################## CapitalOne - National CTR as %s%% #######################'%arr_i[i])
    print('The optimal revenue is')
    print(prob.value)
    print('The optimal allocation in the sports category is')
    print(x1.value)
    print('The optimal allocation in the national category is')
    print(x2.value)
        
    #Appending to matrix
    cap1_n = cap1_n.append([[arr_i[i],prob.value]]) 

######################## CapitalOne - National CTR as 1.5% #######################
The optimal revenue is
511199.99999992014
The optimal allocation in the sports category is
[[2999999.99999952]
 [ 999999.99999976]
 [1000000.00000101]
 [ 999999.99999954]]
The optimal allocation in the national category is
[[1000000.00000017]
 [1000000.00000024]
 [1999999.99999899]
 [1000000.00000046]]
######################## CapitalOne - National CTR as 1.6% #######################
The optimal revenue is
513599.99999997136
The optimal allocation in the sports category is
[[2999999.99999974]
 [ 999999.99999963]
 [1000000.00000011]
 [1000000.00000045]]
The optimal allocation in the national category is
[[1000000.00000011]
 [1000000.00000037]
 [1999999.99999989]
 [ 999999.99999955]]
######################## CapitalOne - National CTR as 1.7000000000000002% #######################
The optimal revenue is
515999.99999925576
The optimal allocation in the sports category is
[[2999999.99998844]
 [ 999999.9999969

In [24]:
#To view the results in a table - sorted the table by top to bottom Revenue 
cap1_n.columns=['National CTR','Optimal Revenue']
cap1_n.sort_values(by=['Optimal Revenue'],ascending=False)

National CTR  Optimal Revenue
0           2.5    535199.999999
0           2.4    532799.999999
0           2.3    530399.999999
0           2.2    527999.999998
0           2.1    525599.999998
0           2.0    523199.999998
0           1.9    520799.999999
0           1.8    518399.999999
0           1.7    515999.999999
0           1.6    513600.000000
0           1.5    511200.000000

In [25]:
#To check the variance in the revenue
print(cap1_n['Optimal Revenue'].mean())
print(cap1_n['Optimal Revenue'].std())

523199.9999988392
7959.8994963720925


#### Results/ Observations/ Recommendations

Below are the highlights of the analysis done -

| Company/Category | CTR - low | CTR - High | Revenue - low | Revenue - High | Variation          
| :---:|:---: | :---:|:---:|:---: |:---:
| Geico - Sports|2%|3%|\$487,200|\$559,200|\$23,880
| Geico - National|0.1%|1.1%|\$506,400|\$530,400|\$7,960
| Delta - Sports|1.5%|2.5%|\$511,200|\$535,200|\$7,960
| Delta - National|0.5%|1.5%|\$511,200|\$535,200|\$7,960
| T-Mobile - Sports|0.5%|1.5%|\$511,200|\$535,200|\$7,960
| T-Mobile - National|2.5%|3.5%|\$499,200|\$547,200|\$15,920
| CapitalOne - Sports|1.5%|2.0%|\$523,200|\$535,200|\$4,490
| CapitalOne - National|1.5%|2.5%|\$511,200|\$535,200|\$7,960

Intuitively, the revenue should be maximum whenever the CTR's are the highest for each of the company and category combination as that would lead to the maximum conversion of the impressions. This is confirmed by the sensitivity analyses performed above. However, since the sensitivity analyses is done one cell at a time (one company-category combination), the revenues vary across each cell. For each combination, whenever the CTR is the maximum, the revenue is higher than the one we get with the static CTRs for the optimal solution at the beginning which is **\$523,200**. Geico Sports section seems to be the most sensitive to the changes in CTR, followed by T-Mobile National. With the higher bound on CTR, they give revenue much higher than the optimal solution, but when the CTR is on the lower bound, the revenue is lower than the optimal solution. Capital One Sports seem to be more stable than other sections, but that can be attributed to the smaller range of CTRs provided.

### Exercise 2 : Operations Excellence

#### For separate optimization on Plant A and B
##### Decision variables
Let $i={1,2}$ be the index for the plants such that
* $i=1$ is Plant A
* $i=2$ is Plant B

Let $j={1,2}$ be the index for the product, such as $j=1$ is the standard product and $j=2$ is the deluxe product.

So $x_{11}$ is the units of standard product from Plant A and $x_{12}$ is the units of deluxe product from Plant A.

##### Objective function
The 1st objective of the firm is to maximize the profit from each plant based on the total units of both the products produced per week. Standard product has a profit contribution of \$10 and deluxe product of \$15.

For Plant A, the 1st objective function is:
\begin{equation}
Profit=10\times x_{11}+15\times x_{12}
\end{equation}

Similarly for Plant B, the 2nd objective function is:
\begin{equation}
Profit=10\times x_{21}+15\times x_{22}
\end{equation}

##### Constraints
For Plant A:-

Based on the processing times (grinding and polishing respectively):
\begin{align}
4\times x_{11}+2\times x_{12}&\leq 80\\
2\times x_{11}+5\times x_{12}&\leq 60\\
\end{align}

Based on raw material (75 kgs cap per week):
\begin{align}
4\times x_{11}+4\times x_{12}&\leq 75\\
\end{align}

Non-negativity constraints are:
\begin{equation}
x_{1j}\geq 0
\end{equation}

For Plant B:-

Based on the processing times (grinding and polishing respectively):
\begin{align}
5\times x_{21}+3\times x_{22}&\leq 60\\
5\times x_{21}+6\times x_{22}&\leq 75\\
\end{align}

Based on raw material (45 kgs cap per week):
\begin{align}
4\times x_{21}+4\times x_{22}&\leq 45\\
\end{align}

Non-negativity constraints are:
\begin{equation}
x_{2j}\geq 0
\end{equation}

##### Solution for Plant A

In [26]:
#Define Variables
x11 = cvx.Int()# Standard Product
x12 = cvx.Int()# Deluxe Product

#Constraints
c1= 4*x11+2*x12<= 80 
c2= 2*x11+5*x12<= 60 
c3= 4*x11+4*x12<= 75 
c4= x11>=0 
c5= x12>=0
con=[c1,c2,c3,c4,c5]

#Objective
OF=10*x11 + 15*x12
objective = cvx.Maximize(OF)

#Solution
prob = cvx.Problem(objective, con)
result = prob.solve()

print('optimal profit')
print(prob.value)

print('Standard production level')
print (x11.value)

print('Deluxe production level')
print (x12.value)

optimal profit
220.00000005215267
Standard production level
10.000000000492811
Deluxe production level
8.000000003148305


##### Solution for Plant B

In [27]:
#Define Variables
x21 = cvx.Int()# Standard Product
x22 = cvx.Int()# Deluxe Product

#Constraints
c1= 5*x21+3*x22<= 60 
c2= 5*x21+6*x22<= 75 
c3= 4*x21+4*x22<= 45 
c4= x21>=0 
c5= x22>=0
con=[c1,c2,c3,c4,c5]

#Objective
OF=10*x21 + 15*x22
objective = cvx.Maximize(OF)

#Solution
prob = cvx.Problem(objective, con)
result = prob.solve()

print('optimal profit')
print(prob.value)

print('Standard production level')
print (x21.value)

print('Deluxe production level')
print (x22.value)

optimal profit
165.00000006681978
Standard production level
2.912859137415921e-09
Deluxe production level
11.000000002512746


#### Solution for Joint optimization

##### Decision variables
Let $i={1,2}$ be the index for the plants such that
* $i=1$ is Plant A
* $i=2$ is Plant B

Let $j={1,2}$ be the index for the product, such as $j=1$ is the standard product and $j=2$ is the deluxe product.

So $x_{11}$ is the units of standard product from Plant A and $x_{12}$ is the units of deluxe product from Plant A.

##### Objective function
The objective of the firm is to maximize the profit together from both plants based on the total units of both the products produced per week. Standard product has a profit contribution of \$10 and deluxe product of \$15.

The objective function is:
\begin{equation}
Profit=10\times \sum_{i=1}^2 x_{i1}+ 15\times \sum_{i=1}^{2} x_{i2}
\end{equation}

##### Constraints
Based on the processing times (grinding and polishing respectively):
\begin{align}
4\times x_{11}+2\times x_{12}&\leq 80\\
2\times x_{11}+5\times x_{12}&\leq 60\\
5\times x_{21}+3\times x_{22}&\leq 60\\
5\times x_{21}+6\times x_{22}&\leq 75\\
\end{align}

Based on raw material (Ttotal 120 kgs cap per week):
\begin{align}
4\times \sum_{i=1}^2\sum_{j=1}^{2} x_{ij}&\leq 120\\
\end{align}

Non-negativity constraints are:
\begin{equation}
x_{ij}\geq 0
\end{equation}

In [28]:
#Based on joint constraint on Raw material
#Total 120 can be distributed based on the final optimal units produced by each plant
#Define Variables
x11 = cvx.Int()# Plant A - Standard Product
x12 = cvx.Int()# Plant A - Deluxe Product
x21 = cvx.Int()# Plant B - Standard Product
x22 = cvx.Int()# Plant B - Deluxe Product

#Constraints
c1= 4*x11+2*x12<= 80 
c2= 2*x11+5*x12<= 60 
c3= 5*x21+3*x22<= 60 
c4= 5*x21+6*x22<= 75 
c5= 4*x11+4*x12+4*x21+4*x22<= 120 
c6= x11>=0 
c7= x12>=0
c8= x21>=0 
c9= x22>=0
con=[c1,c2,c3,c4,c5,c6,c7,c8,c9]

#Objective
OF=10*x11 + 15*x12 + 10*x21 + 15*x22
objective = cvx.Maximize(OF)

#Solution
prob = cvx.Problem(objective, con)
result = prob.solve()

print('optimal profit')
print(prob.value)

print('Plant A - Standard production level')
print (x11.value)

print('Plant A - Deluxe production level')
print (x12.value)

print('Plant B - Standard production level')
print (x21.value)

print('Plant B - Deluxe production level')
print (x22.value)

optimal profit
399.99999992548265
Plant A - Standard production level
9.999999999202224
Plant A - Deluxe production level
7.9999999965348945
Plant B - Standard production level
-1.4098915363594164e-09
Plant B - Deluxe production level
11.999999999969063


#### Results/ Observations/ Recommendations

When the optimization is done for each factory separately, the optimal profit level for Plant A is \$220 and Plant B is \$165, making the total profit as \$385. Also, based on the results, Plant A should produce 10 units of Standard product and 8 units of Deluxe. Plant B should produce 11 units of Deluxe and should not produce any Standard product. Based on joint optimization (based on total 120kgs of raw), a profit level of \$400 can be achieved with Plant A producing 10 units of Standard and 8 units of Deluxe, and Plant B producing 12 units of Deluxe. So, compared to the separate optimization, the joint one allows 1 extra unit to be produced by Plant B and increasing the profit level by \$15.
Recommendations to the CEO to improve the firm's opertaions -
* The firm should go with the results of the joint optimization to maximize the profit. The raw materials to the plants should not be constrained individually, but distributed weekly based on the optimal production level of each plant.
* Plant B should not produce any Standard products and efforts should be focused solely on producing Deluxe products. The Standard products require much more grinding/polishing time in Plant B (due to older machinery) than Plant A.
* In future, if the firm invests in replacing or upgrading the machinery for Plant B, we can run another optimization model to maximize the overall profit level based on the new unit processing times.

### Exercise 3 : Hotel La Quinta Motor Inns (LQM Profitability Analysis)

LQM is a middle-sized hotel chain that is considering expanding to more locations. LQM used data on 75 existing inn locations to build a linear regression model to predict “Profitability”, computed at the operating margin, or earnings before interest and taxes divided by total revenue. They tried many independent variables and came up with the final model

***Profitability = 39.05 - (5.41 x State Population per Inn) + (5.86 x Price of the Inn) - (3.09 x Square Root of the Median Income in the area) + (1.75 x College Students in the Area)***

All independent variables are significant and were normalized to have mean zero and standard deviation 1.

#### Question 1
According to the regression equation given above, which variables positively affect Profitability? Which variables negatively affect Profitability? Does this intuitively make sense? Why?

**Answer:**
According to the equation above, "Price of the Inn" and "College Students in the Area" positively affect Profitability, whereas "State Population per Inn" and "Square Root of the Meadian Income in the area" negatively affect Profitability. 

Some of it does not make sense intuitively. For example, higher price of the Inn doesn't always imply that it will result in high profitability. With the rise of affordable and accessible hotel business (eg.-AirBnB), and knowing that the LQM is a middle-sized hotel chain at present, high prices could lead to lower sales due to customer perception. If the prices are high, then cost of maintainence and service should also be high and that could lead to lower profitability. Other than the hotel prices for LQM, no other cost is mentioned. 

Also, both prices and number of College students variables seem confounding. College students generally look for affordable hotels. However, if we look at only the number of college students, then it makes sense that most of them would not have their own residence in the area and their family and friends would likely stay in the hotel in the same area when they visit.

State population per Inn is not straightforward. Rather, it is counterintuitive because most of the luxury hotels which are profitable are located at the center of large cities. If we take examples of cities like San Francisco and New York which are in densely populated states, yet the hotel business is doing pretty good because tourists all over the world come to visit these places. The square root of median income has an inverse relationship which implies with increase in median income, profitability decreases and then starts saturating after a point.

#### Question 2

Using this regression equation, LQM created a spreadsheet model to predict profitability. LQM collected data for several locations in California, which is provided in the excel spreadsheet on Canvas “LQM”. Using this spreadsheet, compute the profitability for each hotel. Which one has the highest profitability? Which one has the lowest profitability?

**Solution:**

In [31]:
# Reading the hotel spreadsheet
# Uncomment below and comment the second one to run on Google colab
#hotel = pd.read_csv("https://raw.githubusercontent.com/p-sama/Optimization-Problems/master/LQM.csv")
hotel = pd.read_csv("LQM.csv")
hotel.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16 entries, 0 to 15
Data columns (total 7 columns):
Hotel                    16 non-null int64
Location                 16 non-null object
Price                    16 non-null int64
Price_norm               16 non-null float64
SqrtMedianIncome_norm    16 non-null float64
ClgStudents_norm         16 non-null float64
PopulationPerInn_norm    16 non-null float64
dtypes: float64(4), int64(2), object(1)
memory usage: 976.0+ bytes


In [32]:
# Calculating profitability based on the regression equation (with normalized variables)
hotel['Profitability'] = 39.05 - (5.41*hotel['PopulationPerInn_norm']) + (5.86*hotel['Price_norm']) - (3.09*hotel['SqrtMedianIncome_norm']) + (1.75*hotel['ClgStudents_norm'])
hotel = hotel.sort_values(by=['Profitability'],ascending=False)
hotel

Hotel                      Location     Price  Price_norm  \
1       2            Fresno, California  10000000        1.70   
5       6        Long Beach, California   8950000        1.40   
0       1            Eureka, California   2925000       -0.30   
2       3            Fresno, California   3750000       -0.07   
3       4            Fresno, California   3500000       -0.14   
13     14  South Lake Tahoe, California   3750000       -0.07   
11     12  South Lake Tahoe, California   2500000       -0.42   
12     13  South Lake Tahoe, California   1975000       -0.57   
9      10  South Lake Tahoe, California   1650000       -0.66   
14     15  South Lake Tahoe, California   1475000       -0.71   
10     11  South Lake Tahoe, California   1125000       -0.81   
15     16  South Lake Tahoe, California    750000       -0.92   
4       5            Fresno, California    325000       -1.04   
8       9       Los Angeles, California   4900000        0.26   
6       7       Los Angeles, California   1950000       -0.58   
7       8       Los Angeles, California   1750000       -0.63   

    SqrtMedianIncome_norm  ClgStudents_norm  PopulationPerInn_norm  \
1                   -0.41              0.31                  -0.47   
5                    0.66              0.48                  -0.56   
0                   -0.81             -0.54                  -1.00   
2                   -0.41              0.31                  -0.47   
3                   -0.41              0.31                  -0.47   
13                  -0.79             -0.59                  -0.43   
11                  -0.79             -0.59                  -0.43   
12                  -0.79             -0.59                  -0.43   
9                   -0.79             -0.59                  -0.43   
14                  -0.79             -0.59                  -0.43   
10                  -0.79             -0.59                  -0.43   
15                  -0.79             -0.59                  -0.43   
4                   -0.41              0.31                  -0.47   
8                    0.17              3.11                   3.11   
6                    0.17              3.11                   3.11   
7                    0.17              3.11                   3.11   

    Profitability  
1         53.3641  
5         49.0842  
0         44.2599  
2         42.9919  
3         42.5817  
13        42.3747  
11        40.3237  
12        39.4447  
9         38.9173  
14        38.6243  
10        38.0383  
15        37.3937  
4         37.3077  
8         28.6657  
6         23.7433  
7         23.4503

Based on the regression equation, Hotel #2 in Fresno, California seems to be the most profitable and Hotel #8 in Los Angeles California the least profitable.

#### Question 3

LQM has a budget of $10 million to spend on hotels. Suppose we used a “greedy” approach where we selected the most profitable hotels until we ran out of budget. So we would start by the most profitable, and then if we had enough budget left, we would buy the hotel we predict to be the second most profitable, and so on.

a. Describe what we would do with this approach, i.e., which hotels would we purchase?

b. What would our total predicted profitability be? (This is the sum of the predicted
profitability of all hotels we purchase.)

c. If we are trying to maximize our total predicted profitability, is this a good approach? How
about if we were trying to maximize the average predicted profitability of the hotels we
select? How about if we had a budget of \$20 million instead of \$10 million?

In [33]:
# Assign ranks based on profitability (highest-1, 2nd highest-2,..so on) for greedy approach
hotel['Profit_Rank'] = hotel['Profitability'].rank(ascending=False)
hotel['Profit_Rank'] = hotel['Profit_Rank'].astype('int')
hotel

Hotel                      Location     Price  Price_norm  \
1       2            Fresno, California  10000000        1.70   
5       6        Long Beach, California   8950000        1.40   
0       1            Eureka, California   2925000       -0.30   
2       3            Fresno, California   3750000       -0.07   
3       4            Fresno, California   3500000       -0.14   
13     14  South Lake Tahoe, California   3750000       -0.07   
11     12  South Lake Tahoe, California   2500000       -0.42   
12     13  South Lake Tahoe, California   1975000       -0.57   
9      10  South Lake Tahoe, California   1650000       -0.66   
14     15  South Lake Tahoe, California   1475000       -0.71   
10     11  South Lake Tahoe, California   1125000       -0.81   
15     16  South Lake Tahoe, California    750000       -0.92   
4       5            Fresno, California    325000       -1.04   
8       9       Los Angeles, California   4900000        0.26   
6       7       Los Angeles, California   1950000       -0.58   
7       8       Los Angeles, California   1750000       -0.63   

    SqrtMedianIncome_norm  ClgStudents_norm  PopulationPerInn_norm  \
1                   -0.41              0.31                  -0.47   
5                    0.66              0.48                  -0.56   
0                   -0.81             -0.54                  -1.00   
2                   -0.41              0.31                  -0.47   
3                   -0.41              0.31                  -0.47   
13                  -0.79             -0.59                  -0.43   
11                  -0.79             -0.59                  -0.43   
12                  -0.79             -0.59                  -0.43   
9                   -0.79             -0.59                  -0.43   
14                  -0.79             -0.59                  -0.43   
10                  -0.79             -0.59                  -0.43   
15                  -0.79             -0.59                  -0.43   
4                   -0.41              0.31                  -0.47   
8                    0.17              3.11                   3.11   
6                    0.17              3.11                   3.11   
7                    0.17              3.11                   3.11   

    Profitability  Profit_Rank  
1         53.3641            1  
5         49.0842            2  
0         44.2599            3  
2         42.9919            4  
3         42.5817            5  
13        42.3747            6  
11        40.3237            7  
12        39.4447            8  
9         38.9173            9  
14        38.6243           10  
10        38.0383           11  
15        37.3937           12  
4         37.3077           13  
8         28.6657           14  
6         23.7433           15  
7         23.4503           16

In [34]:
# Define the Greedy algo function to add the list of most to least profitable hotels until we exhaust the budget
def greedy_app(budget):
    hotel_list = pd.DataFrame()
    for i in range(1,16):
        price = hotel.loc[hotel['Profit_Rank']==i,'Price'].iloc[0]
        hid = hotel.loc[hotel['Profit_Rank']==i,'Hotel'].iloc[0]
        location = hotel.loc[hotel['Profit_Rank']==i,'Location'].iloc[0]
        profit = hotel.loc[hotel['Profit_Rank']==i,'Profitability'].iloc[0]
        
        if (budget >= price):
            hotel_list = hotel_list.append([[hid, location, price, profit, i]]) 
            budget = budget - price
    
    hotel_list.columns = ['Hotel','Location','Price','Profitability','Profit_Rank']
    return hotel_list

In [35]:
#Run the function with $10Mn budget
greedy_app(10000000)

Hotel            Location     Price  Profitability  Profit_Rank
0      2  Fresno, California  10000000        53.3641            1

In [36]:
#Run the function with $20Mn budget
greedy_app(20000000)

Hotel                      Location     Price  Profitability  Profit_Rank
0      2            Fresno, California  10000000        53.3641            1
0      6        Long Beach, California   8950000        49.0842            2
0     16  South Lake Tahoe, California    750000        37.3937           12

In [37]:
greedy_app(20000000).Profitability.sum()

139.84199999999998

If we use the greedy approach on total profitability with \$10 million budget, we would purchase the most profitable hotel (based on the regression equation) - Hotel #2 in Fresno California and our total predicted profitability (and average profitability) would be \$53.3641 (normalized).

However, if we use the same approach with \$20 million budget, we would purchase three hotels - Hotel #2 in Fresno California, #6 in Long Beach, California and #16 in South Lake Tahoe, California and our total predicted profitability would be \$139.842 and average profitability would be \~$46 (normalized).

This is not the best approach as we exhaust the budget with very few hotels. We could select a larger portfolio of hotels with lesser prices and maximize the total profitability. So, average profitability is not a good metric to look at for making such decisions. If we use total profitability, then greedy approach would always end up being the best solution because average profitablity like in this case with \$10Mn, average profitability will be \$53 and even if you increase the budget and select more hotels with lesser profit, they will bring down your average. So, total profitability should be used as the metric and Optimization is the best solution to such problems to select the best portfolio for the investment.

#### Question 4

Now, build an optimization model to select hotels given the \$10 million budget.

a. Write out the optimization problem. Make sure to detail the decision variables, the objective function and the constraints.

b. What is the optimal solution? Give the values of the decision variables and the optimal objective function value.

c. Does the optimal solution make sense intuitively? How does it compared to the greedy solution?

In [38]:
# Sorting the data back to original index to maintain the sequence for further indexing
hotel = hotel.sort_index()
hotel

Hotel                      Location     Price  Price_norm  \
0       1            Eureka, California   2925000       -0.30   
1       2            Fresno, California  10000000        1.70   
2       3            Fresno, California   3750000       -0.07   
3       4            Fresno, California   3500000       -0.14   
4       5            Fresno, California    325000       -1.04   
5       6        Long Beach, California   8950000        1.40   
6       7       Los Angeles, California   1950000       -0.58   
7       8       Los Angeles, California   1750000       -0.63   
8       9       Los Angeles, California   4900000        0.26   
9      10  South Lake Tahoe, California   1650000       -0.66   
10     11  South Lake Tahoe, California   1125000       -0.81   
11     12  South Lake Tahoe, California   2500000       -0.42   
12     13  South Lake Tahoe, California   1975000       -0.57   
13     14  South Lake Tahoe, California   3750000       -0.07   
14     15  South Lake Tahoe, California   1475000       -0.71   
15     16  South Lake Tahoe, California    750000       -0.92   

    SqrtMedianIncome_norm  ClgStudents_norm  PopulationPerInn_norm  \
0                   -0.81             -0.54                  -1.00   
1                   -0.41              0.31                  -0.47   
2                   -0.41              0.31                  -0.47   
3                   -0.41              0.31                  -0.47   
4                   -0.41              0.31                  -0.47   
5                    0.66              0.48                  -0.56   
6                    0.17              3.11                   3.11   
7                    0.17              3.11                   3.11   
8                    0.17              3.11                   3.11   
9                   -0.79             -0.59                  -0.43   
10                  -0.79             -0.59                  -0.43   
11                  -0.79             -0.59                  -0.43   
12                  -0.79             -0.59                  -0.43   
13                  -0.79             -0.59                  -0.43   
14                  -0.79             -0.59                  -0.43   
15                  -0.79             -0.59                  -0.43   

    Profitability  Profit_Rank  
0         44.2599            3  
1         53.3641            1  
2         42.9919            4  
3         42.5817            5  
4         37.3077           13  
5         49.0842            2  
6         23.7433           15  
7         23.4503           16  
8         28.6657           14  
9         38.9173            9  
10        38.0383           11  
11        40.3237            7  
12        39.4447            8  
13        42.3747            6  
14        38.6243           10  
15        37.3937           12

**Optimization Model**

**Decision variables**

Let y be a binary variable representing the decision to buy a hotel or not.

Let $i={1,2,..,16}$ be the index for each hotel, so based on the input data sequence - $y_{1}$ is the decision to buy that hotel or not (binary 1-buy, 0-don't buy) an so on.

**Objective function**

Let z be the predicted profitability for each hotel, so, $z_{1}$ is the proftability for the hotel #1 in Eureka, California and so on.
The objective is to maximize the profit from the portfolio of selected hotels to buy.

The objective function is:
\begin{equation}
Profit=\sum_{i=1}^{16} z_{i}\times y_{i}
\end{equation}

**Constraints**

Let x be the array containing the Prices, so, $x_{1}$ is the price for the hotel #1 in Eureka, California and so on.
The total budget should be \$10 Mn and y has to be binary (1/0) based on the decision to buy (or not):
\begin{align}
\sum_{i=1}^{16} x_{i}\times y_{i}&\leq 10000000\\
y\in&{0,1}&\\
\end{align}

From the data, we know that price and profit are not negative, but to be consistent (profit could be negative in some cases) we put non-negativity constraints as follows (although these are not used in the code):
\begin{equation}
x_{i}\geq 0\\
z_{i}\geq 0
\end{equation}

In [49]:
#Define Variables
n = 16 # Number of hotels
y = cvx.Bool(n)

# from the data
x = hotel.iloc[:,2].values 
z = hotel.iloc[:,7].values

#Constraints
c1= sum_entries(x*y)<= 10000000
c2= sum_entries(y)<= n
con=[c1,c2]

#Objective
OF=sum_entries(z*y)
objective = cvx.Maximize(OF)

#Solution
prob = cvx.Problem(objective, con)
result = prob.solve(solver=GLPK_MI)

print('Optimal profit')
print(prob.value)

print('Decision to buy each hotel')
print (y.value)

Optimal profit
270.04970000000003
Decision to buy each hotel
[[0.]
 [0.]
 [0.]
 [0.]
 [1.]
 [0.]
 [0.]
 [0.]
 [0.]
 [1.]
 [1.]
 [1.]
 [1.]
 [0.]
 [1.]
 [1.]]


In [42]:
import numpy as np

buy_dec = pd.DataFrame(np.array((y.value).T)[0])
buy_dec.columns = ['buy_dec']
hotel_model1 = pd.concat([hotel,buy_dec],axis=1)
print("Total budget used:");print(hotel_model1.loc[hotel_model1['buy_dec']>0,'Price'].sum())
print("The hotels to be bought:")
hotel_model1.loc[hotel_model1['buy_dec']>0]

Total budget used:
9800000
The hotels to be bought:


Hotel                      Location    Price  Price_norm  \
4       5            Fresno, California   325000       -1.04   
9      10  South Lake Tahoe, California  1650000       -0.66   
10     11  South Lake Tahoe, California  1125000       -0.81   
11     12  South Lake Tahoe, California  2500000       -0.42   
12     13  South Lake Tahoe, California  1975000       -0.57   
14     15  South Lake Tahoe, California  1475000       -0.71   
15     16  South Lake Tahoe, California   750000       -0.92   

    SqrtMedianIncome_norm  ClgStudents_norm  PopulationPerInn_norm  \
4                   -0.41              0.31                  -0.47   
9                   -0.79             -0.59                  -0.43   
10                  -0.79             -0.59                  -0.43   
11                  -0.79             -0.59                  -0.43   
12                  -0.79             -0.59                  -0.43   
14                  -0.79             -0.59                  -0.43   
15                  -0.79             -0.59                  -0.43   

    Profitability  Profit_Rank  buy_dec  
4         37.3077           13      1.0  
9         38.9173            9      1.0  
10        38.0383           11      1.0  
11        40.3237            7      1.0  
12        39.4447            8      1.0  
14        38.6243           10      1.0  
15        37.3937           12      1.0

As expected, the optimization model seems to be much better than the greedy approach. In the greedy approach, we exhausted all the budget in buying just one hotel with total profitability of \$53, whereas in this approach, we achieve a much higher profit level of \$270 and still left with \$0.2Mn in the budget.

#### Question 5

LQM thinks that buying too many hotels in one city is probably not a good idea and would prefer to diversify across as many cities as possible. Add constraint(s) to your model to limit the number of hotels purchased in any city to at most 2.

a. What are the constraints that you need to add to the model? Intuitively, do you expect the new optimal objective function value to be larger, smaller or the same as before?

b. Write the new optimization model.

c. Solve the new model. Give the values of the decision variables and the optimal objective
function value. How does this compare to the previous solution?

**Additional Constraint for the 2nd Optimization model**

The constraint needs to be on the y values - group of y values, representing the hotels in the same city, should be maximum 2. Intuitively, by taking a glance at the predicted profitability, it seems that the optimal objective function value will be lower. If you look at the previous solution, a lot of hotels were chosen in the same city (South Lake Tahoe), which have slightly better profitability than other cities like LA. So, if there is a constraint then maybe the next set of hotels could be from LA which will give lower profit. There are some hotels in other cities like Fresno and Long Beach where the profitability is very high but since the price is also very high, it's less likely for the optimization model to choose them so that the budget is not exhausted early (like in the greedy approach).

Thus, the additional constraints on the cities of Fresno, LA and South Lake Tahoe are:

\begin{align}
y_{1}+y_{2}+y_{3}+y_{4}&\leq 2\\
y_{6}+y_{7}+y_{8}&\leq 2\\
y_{9}+y_{10}+y_{11}+y_{12}+y_{13}+y_{14}+y_{15}&\leq 2\\
\end{align}

In [43]:
#Define Variables
n = 16 # Number of hotels
y = cvx.Bool(n)

# from the data
x = hotel.iloc[:,2].values
z = hotel.iloc[:,7].values

#Constraints
c1= sum_entries(x*y)<= 10000000
c2= sum_entries(y)<= n
c3= sum_entries(y[1:5])<= 2
c4= sum_entries(y[6:9])<= 2
c5= sum_entries(y[9:])<= 2
con=[c1,c2,c3,c4,c5]

#Objective
OF=sum_entries(z*y)
objective = cvx.Maximize(OF)

#Solution
prob = cvx.Problem(objective, con)
result = prob.solve(solver=GLPK_MI)

print('Optimal profit')
print(prob.value)

print('Decision to buy each hotel')
print (y.value)

Optimal profit
205.71679999999992
Decision to buy each hotel
[[1.]
 [0.]
 [0.]
 [0.]
 [1.]
 [0.]
 [1.]
 [1.]
 [0.]
 [1.]
 [1.]
 [0.]
 [0.]
 [0.]
 [0.]
 [0.]]


In [44]:
buy_dec = pd.DataFrame(np.array((y.value).T)[0])
buy_dec.columns = ['buy_dec']
hotel_model2 = pd.concat([hotel,buy_dec],axis=1)
print("Total budget used:");print(hotel_model2.loc[hotel_model2['buy_dec']>0,'Price'].sum())
print("The hotels to be bought:")
hotel_model2.loc[hotel_model2['buy_dec']>0]

Total budget used:
9725000
The hotels to be bought:


Hotel                      Location    Price  Price_norm  \
0       1            Eureka, California  2925000       -0.30   
4       5            Fresno, California   325000       -1.04   
6       7       Los Angeles, California  1950000       -0.58   
7       8       Los Angeles, California  1750000       -0.63   
9      10  South Lake Tahoe, California  1650000       -0.66   
10     11  South Lake Tahoe, California  1125000       -0.81   

    SqrtMedianIncome_norm  ClgStudents_norm  PopulationPerInn_norm  \
0                   -0.81             -0.54                  -1.00   
4                   -0.41              0.31                  -0.47   
6                    0.17              3.11                   3.11   
7                    0.17              3.11                   3.11   
9                   -0.79             -0.59                  -0.43   
10                  -0.79             -0.59                  -0.43   

    Profitability  Profit_Rank  buy_dec  
0         44.2599            3      1.0  
4         37.3077           13      1.0  
6         23.7433           15      1.0  
7         23.4503           16      1.0  
9         38.9173            9      1.0  
10        38.0383           11      1.0

With the additional constraint, the total profitability goes down slightly and a different portfolio of hotels is selected. We had sensed this while designing the model.

#### Results/ Observations/ Recommendations

Following results will be presented to the LQM executives -

| Approach | Scenario | Number of Hotelts bought | Total Profitability | Budget exhausted         
| :---:|:---: | :---:|:---:|:---:
| Greedy Algorithm|\$10Mn Budget|1|\$53.36|\$10,000,000
| Greedy Algorithm|\$20Mn Budget|3|\$139.84|\$19,700,000
| Optimization Model|\$10Mn Budget|7|\$270.05|\$9,800,000
| Optimization Model|\$10Mn Budget & no city with >2 hotels|6|\$205.72|\$9,725,000


Based on the above results, I would recommend LQM to base their decision on either of the optimization solutions. The final decision would depend on whether the company is looking for high risk-high return investment or low risk-low return option. The first model gives higher profitability but selects most of the hotels in the same city, whereas the second model gives slightly lower profitability but since the portfolio is diversified, risk of losing business due to unforeseen circumstances in future (lower demand for tourism, seasonality, cheaper competition in the same area, natural calamities, etc.) would be low.

Recommendations on improving the regression equation - Adding information on Competitor hotels and substitutes (like AirBnb) - (Distance, number, prices), how old is the hotel, space - number of rooms and other common areas, accessibility of the hotel

Recommendations on improving the optimization model - To improve the optimization model, additional constraints can be considered on some of the additional variables selected for the regression model improvement. Also, the above optimization models are deterministic in nature, but there are many random uncertain events that could take place and should be ideally accounted for in some form or the other. Monter Carlo simulation is particularly helpful in such situations as it can run several scenarios based on pre-determined factors which are driven by their assigned probabilities (Reference: http://www.hetherley.com/how-monte-carlo-simulations-improve-hotel-investment-models/)